File: app/src/main/java/com/example/loginsignupapi/Retrofit/JsonApi.java
package com.example.loginsignupapi.Retrofit;

import com.example.loginsignupapi.Models.ModelCheckPhoneEmail;
import com.example.loginsignupapi.Models.ModelUserLogin;
import com.example.loginsignupapi.Models.ModelUserRegister;


import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface JsonApi
{

    @FormUrlEncoded
    @POST("checkPhoneNumberAndEmail")
    Call<ModelCheckPhoneEmail> checkNumberAndEmail(@Field("phoneNumber") String phoneNumber,
                                                   @Field("email") String email);
    @FormUrlEncoded
    @POST("userRegister")
    Call<ModelUserRegister> userRegister(@Field("name") String name,
                                         @Field("email") String email,
                                         @Field("phone") String phone,
                                         @Field("password") String password,
                                         @Field("device_type") String device_type,
                                         @Field("reg_id") String reg_id);
    @FormUrlEncoded
    @POST("userLogin")
    Call<ModelUserLogin> userLogin(@Field("emailPhone") String emailPhone,
                                   @Field("password") String password,
                                   @Field("reg_id") String reg_id,
                                   @Field("device_type") String device_type);
}

File: app/src/main/java/com/example/loginsignupapi/Retrofit/JsonApi.kt
package com.example.loginsignupapi.Retrofit

import com.example.loginsignupapi.Models.*
import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.*

interface JsonApi {
    @FormUrlEncoded
    @POST("checkPhoneNumberAndEmail")
    fun checkNumberAndEmail(@Field("phoneNumber") phoneNumber: String?,
                            @Field("email") email: String?): Call<ModelCheckPhoneEmail?>?

    @FormUrlEncoded
    @POST("userRegister")
    fun userRegister(@Field("name") name: String?,
                     @Field("email") email: String?,
                     @Field("phone") phone: String?,
                     @Field("password") password: String?,
                     @Field("device_type") device_type: String?,
                     @Field("reg_id") reg_id: String?): Call<ModelUserRegister?>?

    @FormUrlEncoded
    @POST("userLogin")
    fun userLogin(@Field("emailPhone") emailPhone: String?,
                  @Field("password") password: String?,
                  @Field("reg_id") reg_id: String?,
                  @Field("device_type") device_type: String?): Call<ModelUserLogin?>?

    @FormUrlEncoded
    @POST("getUserProfile")
    fun userProfile(@Field("userId") userId: String?): Call<ModelGetUserProfile?>?

    @Multipart
    @POST("userUpdateProfile")
    fun updateProfile(@Part("userId") userId: RequestBody?,
                      @Part("name") name: RequestBody?,
                      @Part("email") email: RequestBody?,
                      @Part("phoneNumber") phoneNumber: RequestBody?,
                      @Part("address") address: RequestBody?,
                      @Part image: MultipartBody.Part?): Call<ModelUpdateProfile?>?
}